pkg/odo/cli/component/common_link.go

```diff
@@ -122,14 +122,35 @@ func (o *commonLinkOptions) run() (err error) {
 		return err
 	}
 
-	if o.operationName == "link" {
-		fmt.Printf("Following environment variables were added to %s component:\n", o.Component())
+	if len(secret.Data) == 0 {
+		log.Infof("There are no secret environment variables to expose within the %s service", o.suppliedName)
 	} else {
-		fmt.Printf("Following environment variables were removed from %s component:\n", o.Component())
-	}
+		if o.operationName == "link" {
+			log.Infof("The below secret environment variables were added to the '%s' component:\n", o.Component())
+		} else {
+			log.Infof("The below secret environment variables were removed from the '%s' component:\n", o.Component())
+		}
+
+		// Output the environment variables
+		for i := range secret.Data {
+			fmt.Printf("· %v\n", i)
+		}
+
+		// Retrieve the first variable to use as an example.
+		// Have to use a range to access the map
+		var exampleEnv string
+		for i := range secret.Data {
```

Comment:
why use a range? why not just `secret.Data[0]` it? I already see `secret.Data == 0` check which means `Data` array should have atleast one element.

Reply:
secret.Data is a map not an array, and that's why you have to use a range to get the first variable.

Reply:
Hehe, felt confusing.